File: src/selection_sort.js
function findSmallest(arr) {
  let smallest = arr[0];
  let smallestIndex = 0;

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] < smallest) {
      smallest = arr[i];
      smallestIndex = i;
    }
  }

  return smallestIndex;
}

function orderBySelection(arr) {
  const newArr = [];
  const originalLength = arr.length;

  for (let i = 0; i < originalLength; i++) {
    let smallestIndex = findSmallest(arr);
    newArr.push(...arr.splice(smallestIndex, 1));
  }

  return newArr;
}

console.log(orderBySelection([5, 3, 6, 2, 10]));

File: src/selection_sort.py
def findSmallest(arr):
  smallest = arr[0]
  smallest_index = 0

  for i in range(1, len(arr)):
    if arr[i] < smallest:
      smallest = arr[i]
      smallest_index = i
  return smallest_index

def orderBySelection(arr):
  newArr = []

  for i in range(len(arr)):
    smallest = findSmallest(arr)
    newArr.append(arr.pop(smallest))
  return newArr

print orderBySelection([5, 3, 6, 2, 10])